akka-docs-dev/rst/scala/code/docs/stream/StreamTcpDocSpec.scala

```diff
@@ -108,7 +101,42 @@ class StreamTcpDocSpec extends AkkaSpec {
 
       connection.handleWith(serverLogic)
     }
+
     //#welcome-banner-chat-server
 
+    val input = new AtomicReference("Hello world" :: "What a lovely day" :: Nil)
+    def readLine(prompt: String): String = {
+      input.get() match {
+        case all @ cmd :: tail if input.compareAndSet(all, tail) ⇒ cmd
+        case _ ⇒ "q"
+      }
+    }
+
+    //#repl-client
+    val connection: OutgoingConnection = StreamTcp().outgoingConnection(localhost)
+
+    val repl = Flow[ByteString]
+      .transform(() => RecipeParseLines.parseLines("\n", maximumLineBytes = 256))
+      .map(text => println("Server: " + text))
+      .map(_ => readLine("> "))
+      //#repl-client
+      .map {
+        case "q" =>
+          sys.shutdown(); println("SHUTTING DOWN NOW")
+        case x   ⇒ x
+      }
+      //#repl-client
+      .map {
+        case "q"  => ByteString("BYE") // translate to server command
+        case text => ByteString(s"$text\n")
+      }
+
+    connection.handleWith(repl)
+    //#repl-client
+
+    serverProbe.expectMsg("Hello world")
+    serverProbe.expectMsg("What a lovely day")
+    import concurrent.duration._
```

Comment:
I don't fully understand why you have to do the sys shutdown in the flow, and not here.